x/authz/client/cli/tx.go

```diff
@@ -0,0 +1,197 @@
+				return err
+			}
+			grantee, err := sdk.AccAddressFromBech32(args[0])
+			if err != nil {
+				return err
+			}
+
+			msgType := args[1]
+
+			var authorization types.Authorization
+			if msgType == (types.SendAuthorization{}.MethodName()) {
+				limit, err := sdk.ParseCoinsNormalized(args[2])
+				if err != nil {
+					return err
+				}
+				authorization = &types.SendAuthorization{
+					SpendLimit: limit,
+				}
+			} else {
+				authorization = types.NewGenericAuthorization(msgType)
+			}
+
+			exp, err := cmd.Flags().GetInt64(FlagExpiration)
+			if err != nil {
+				return err
+			}
+
+			msg, err := types.NewMsgGrantAuthorization(clientCtx.GetFromAddress(), grantee, authorization, time.Unix(exp, 0))
+			if err != nil {
+				return err
+			}
+
+			svcMsgClientConn := &helpers.ServiceMsgClientConn{}
+			authzMsgClient := types.NewMsgClient(svcMsgClientConn)
+			_, err = authzMsgClient.GrantAuthorization(context.Background(), msg)
+			if err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), svcMsgClientConn.GetMsgs()...)
+
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	cmd.Flags().Int64(FlagExpiration, time.Now().AddDate(1, 0, 0).Unix(), "The Unix timestamp. Default is one year.")
+	return cmd
+}
+
+func NewCmdRevokeAuthorization() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "revoke [grantee_address] [msg_type] --from=[granter_address]",
+		Short: "revoke authorization",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`revoke authorization from a granter to a grantee:
+Example:
+ $ %s tx %s revoke cosmos1skj.. %s --from=cosmos1skj..
+			`, version.AppName, types.ModuleName, types.SendAuthorization{}.MethodName()),
+		),
+		Args: cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			grantee, err := sdk.AccAddressFromBech32(args[0])
+			if err != nil {
+				return err
+			}
+
+			granter := clientCtx.GetFromAddress()
+
+			msgAuthorized := args[1]
+
+			msg := types.NewMsgRevokeAuthorization(granter, grantee, msgAuthorized)
+
+			svcMsgClientConn := &helpers.ServiceMsgClientConn{}
+			authzMsgClient := types.NewMsgClient(svcMsgClientConn)
+			_, err = authzMsgClient.RevokeAuthorization(context.Background(), &msg)
+			if err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), svcMsgClientConn.GetMsgs()...)
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+func NewCmdExecAuthorization() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "exec [msg_tx_json_file] --from [grantee]",
+		Short: "execute tx on behalf of granter account",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`execute tx on behalf of granter account:
+Example:
+ $ %s tx %s exec tx.json --from grantee
+ $ %s tx bank send <granter> <recipient> --from <granter> --chain-id <chain-id> --generate-only > tx.json && %s tx %s exec tx.json --from grantee
```

Comment:
To make this command work, I believe we need https://github.com/cosmos/cosmos-sdk/issues/8306